src/librustc_typeck/check/demand.rs

```diff
@@ -210,7 +211,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
     /// ```
     /// opt.map(|arg| { takes_ref(arg) });
     /// ```
-    fn can_use_as_ref(&self, expr: &hir::Expr) -> Option<(Span, &'static str, String)> {
+    fn can_use_as_ref(
+        &self,
+        expr: &hir::Expr,
+        cm: &SourceMap
```

Comment:
You don't need this here, you can use `self.sess().source_map()` in this method instead of passing it as an argument.